### PR TITLE
Resin uboot on beaglebone

### PR DESCRIPTION
**Pull request by agherzan**
Connected to https://github.com/resin-os/meta-resin/pull/584

### CODE REVIEW

File: layers/meta-resin-beaglebone/recipes-bsp/u-boot/u-boot-ti-staging/beaglebone-integrate-with-resin-configuration.patch [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
++++ b/include/configs/am335x_evm.h
+@@ -96,8 +96,11 @@
+ 	"fi;"	\
+ 	"run findfdt; " \
+ 	"run init_console; " \
+-	"run envboot; " \
+-	"run distro_bootcmd"
++	"setenv resin_kernel_load_addr ${loadaddr}; "\
++	"run resin_set_kernel_root; "\
++	"setenv mmcdev ${resin_mmcdev}; "\
```
Comment (floion) [inline layers/meta-resin-beaglebone/recipes-bsp/u-boot/u-boot-ti-staging/beaglebone-integrate-with-resin-configuration.patch]: I think in this patch you are supposed to actually define resin_mmcdev. Above you redefine mmcdev based on a non-existing resin_mmcdev?

Comment (agherzan) [inline layers/meta-resin-beaglebone/recipes-bsp/u-boot/u-boot-ti-staging/beaglebone-integrate-with-resin-configuration.patch]: I was a bug in the independent patch. I"ll fix.

File: layers/meta-resin-beaglebone/recipes-support/resin-init/resin-init-flasher.bbappend [hunk 2]
```diff
@@ -1,9 +1,2 @@
 # Flash to internal eMMC - kernel enumerates this, when SD card is in, as mmcblk1
 INTERNAL_DEVICE_KERNEL_beaglebone = "mmcblk1"
-
-# the name of the bootloader config file to be written to internal media
-INTERNAL_DEVICE_BOOTLOADER_CONFIG_beaglebone = "uEnv.txt_internal"
-
-# the pathname (relative to /boot/) where the internal bootloader config file should reside in
-# (currently assuming this is on partition 1 of the internal media)
-INTERNAL_DEVICE_BOOTLOADER_CONFIG_PATH_beaglebone = "/uEnv.txt"
```
Comment (floion) [inline layers/meta-resin-beaglebone/recipes-support/resin-init/resin-init-flasher.bbappend]: Shall we also remove now the code in the flasher script that handles these vars?

Comment (agherzan) [inline layers/meta-resin-beaglebone/recipes-support/resin-init/resin-init-flasher.bbappend]: I thought about it. Let's leave it for now for boards that were not yet migrated to resin-uboot.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,8 @@
 Change log
 -----------
 
+* Implement resin-uboot specification 2.0 for beaglebone [Andrei]
+
 # v2.0.0-beta12.rev2 - 2017-02-27
```
Comment (floion) [inline CHANGELOG.md]: As pointed out by Michal, this PR is incomplete

Comment (agherzan) [inline CHANGELOG.md]: @michal-mazurek @floion Fixed the uEnv issue.

**Comment by floion** [comment]
As pointed out by Shaun, this PR should also incorporate the needed changes so that users can load dtbs.
Either through the old uEnv.txt or through the new resinOS_uEnv.txt (as this gets imported in the uboot env so it can be an alternative)

**Comment by floion** [comment]
@agherzan care to reply to Michal's question?

**Comment by agherzan** [comment]
Hi @michal-mazurek . The empty `uEnv.txt_internal` is kept because it will be pushed by the flasher in the internal device as an empty uEnv.txt . This file is used by the bootloader for various operations (ex load a specific dtb).

**Pull request merged**